src/scheduler.cc

```diff
@@ -145,7 +145,16 @@ Status SchedulerService::NotifyTaskCompleted(ServerContext* context, const Notif
     avail_workers_.push_back(request->workerid());
   }
   if (!request->task_succeeded()) {
-    RAY_LOG(RAY_FATAL, "The task on worker " << request->workerid() << " threw an exception with the following error message: " << request->error_message());
+    std::string worker_address;
+    {
+      std::lock_guard<std::mutex> workers_lock(workers_lock_);
+      worker_address = workers_[request->workerid()].worker_address;
+    }
+    {
+      std::lock_guard<std::mutex> failed_workers_lock(failed_workers_lock_);
+      failed_workers_.push_back(worker_address);
+    }
+    RAY_LOG(RAY_FATAL, "The task on worker " << request->workerid() << " at address " << worker_address << " threw an exception with the following error message: " << request->error_message());
```

Comment:
We'd probably like to avoid throwing a fatal error here (we won't be able to test your code if the scheduler dies whenever a task fails, since the task failure has to happen before we can return any information about failed tasks).

However, we may have to update some of the data structures in the event of the failure (e.g., the object table and other things). I'll have to give this more thought..